internal/services/watchlist/watchlist.go

```diff
@@ -0,0 +1,89 @@
+package watchlist
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"errors"
+)
+
+func CreateWatchlist(Req CreateRequest) (Res CreateResponse, err error) {
+	var user model.Users
+	err = config.DB.Table("users").Where("user_id = ?", Req.UserId).First(&user).Error
+	if err != nil {
+		return Res, errors.New("user not found")
+	}
+	var wl model.Watchlist
+	err = config.DB.Table("watchlist").Where("name = ?", Req.WatchlistName).First(&wl).Error
+	if err == nil {
+		return Res, errors.New("watchlist name already exists")
+	}
+	wl.Name = Req.WatchlistName
+	wl.Userid = Req.UserId
+	err = config.DB.Table("watchlist").Create(&wl).Error
+	if err != nil {
+		return Res, errors.New("failed to create watchlist")
+	}
+	Res.Message = "Watchlist created"
+	Res.WatchlistId = wl.Id
+
+	return Res, nil
+}
+
+func AddStockEntry(Req AddStockRequest, watchlistId int) (Res AddStockResponse, err error) {
+	var wl model.Watchlist
+	err = config.DB.Table("watchlist").Where("user_id = ?", Req.UserId).First(&wl).Error
+	if err != nil {
+		return Res, errors.New("user not found")
+	}
+	err = config.DB.Table("watchlist").Where("watchlist_id = ? ", watchlistId).First(&wl).Error
+	if err != nil {
+		return Res, errors.New("watchlist not found")
+	}
+	var uwl model.UserWatchlist
+
+	err = config.DB.Table("user_watchlist").Where("stock_name = ? AND watchlist_id = ?", Req.StockName, watchlistId).First(&uwl).Error
```

Comment:
there can be same stock under same watchlist name for different users right?

Reply:
thats why i have written watchlist_id instead of watchlist_name

Reply:
not resolved yet.